Library/Homebrew/utils/fork.rb

```diff
@@ -51,7 +53,7 @@ def self.safe_fork
           # to rescue them further down.
           if e.is_a?(ErrorDuringExecution)
             error_hash["cmd"] = e.cmd
-            error_hash["status"] = e.status.exitstatus
+            error_hash["status"] = Base64.encode64(Marshal.dump(e.status))
```

Comment:
```suggestion
            error_hash["status"] = e.status&.exitstatus
```

and instead never assume that this is non-`nil`.

Reply:
Fair enough on Marshal, but the problem isn't `status` being nil. It's `exitstatus` being nil. If `exitstatus` is nil, then we need to use `termsig` (see `ErrorDuringExecution`). So will have to change around the API a bit.

Reply:
Changing API around seems fine to me. As long as we assume non-zero status code I don't think preserving this is super critical (but I'll happily see a more comprehensive fix)